cli.go

```diff
@@ -0,0 +1,580 @@
+				} else {
+					runOptions.config = runOptions.setupOptions.configPath
+				}
+				if ctx.IsSet("data") {
+					runOptions.dataStore = ctx.String("data")
+				}
+				if runOptions.Config.ServerCert != "" &&
+					runOptions.setupOptions.serverCert == "" {
+					runOptions.setupOptions.serverCert = runOptions.Config.ServerCert
+				} else {
+					runOptions.Config.ServerCert = runOptions.setupOptions.serverCert
+				}
+				return runOptions.handleCLIOptions(ctx)
+			},
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:        "config, c",
+					Destination: &runOptions.setupOptions.configPath,
+					Value:       defaultConfFile,
+					Usage:       "`PATH` to configuration file."},
+				cli.StringFlag{
+					Name:  "data, d",
+					Usage: "Mender state data `DIR`ECTORY path.",
+					Value: defaultDataStore},
+				cli.StringFlag{
+					Name:        "device-type, t",
+					Destination: &runOptions.setupOptions.deviceType,
+					Usage:       "Name of the device `type`."},
+				cli.StringFlag{
+					Name:        "username, u",
+					Destination: &runOptions.setupOptions.username,
+					Usage:       "User `E-Mail` at hosted.mender.io."},
+				cli.StringFlag{
+					Name:        "password, p",
+					Destination: &runOptions.setupOptions.password,
+					Usage:       "User `PASSWORD` at hosted.mender.io."},
+				cli.StringFlag{
+					Name:        "server-url, s",
+					Destination: &runOptions.setupOptions.serverURL,
+					Usage:       "`URL` to Mender server.",
+					Value:       "https://docker.mender.io"},
+				cli.StringFlag{
+					Name:        "server-ip, 4",
+					Destination: &runOptions.setupOptions.serverIP,
+					Usage:       "Server ip address."},
+				cli.StringFlag{
+					Name:        "server-cert, E",
+					Destination: &runOptions.setupOptions.serverCert,
+					Usage:       "`PATH` to trusted server certificates"},
+				cli.StringFlag{
+					Name:        "tenant-token, T",
+					Destination: &runOptions.setupOptions.tenantToken,
+					Usage:       "Hosted Mender tenanant `token`"},
+				cli.IntFlag{
+					Name:        "inventory-poll, I",
+					Destination: &runOptions.setupOptions.invPollInterval,
+					Usage:       "Inventory poll interval in `sec`onds."},
+				cli.IntFlag{
+					Name:        "retry-poll, R",
+					Destination: &runOptions.setupOptions.retryPollInterval,
+					Usage:       "Retry poll interval in `sec`onds."},
+				cli.IntFlag{
+					Name:        "update-poll, P",
+					Destination: &runOptions.setupOptions.updatePollInterval,
+					Usage:       "Update poll interval in `sec`onds."},
+				cli.BoolFlag{
+					Name:        "hosted-mender, H",
+					Destination: &runOptions.setupOptions.hostedMender,
+					Usage:       "Setup device towards Hosted Mender."},
+				cli.BoolFlag{
+					Name:        "demo, D",
+					Destination: &runOptions.setupOptions.demo,
+					Usage:       "Use demo configuration."},
+				cli.BoolFlag{
+					Name:  "run-daemon, r",
+					Usage: "Run daemon after setup."},
+			},
```

Comment:
Many different options leading to utilize lower and upper case of the same letters. I think that for most of them are not really helping in clarifying the usage (`-4` for ip, `-I` for inventory?).

Except for the most important ones (IMO demo, daemon and device-type) I think we should keep the rest only with the long version with double `--`. A command produced with so many small flags will be totally unreadable.